rabbitmq/include/userver/urabbitmq/client.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <memory>
+
+#include <userver/clients/dns/resolver_fwd.hpp>
+#include <userver/utils/fast_pimpl.hpp>
+
+#include <userver/urabbitmq/client_settings.hpp>
+
+USERVER_NAMESPACE_BEGIN
+
+namespace urabbitmq {
+
+class AdminChannel;
+class Channel;
+class ReliableChannel;
+class ConsumerBase;
+class ClientImpl;
+
+/// Interface for communicating with a RabbitMQ cluster.
+class Client final : public std::enable_shared_from_this<Client> {
```

Comment:
For classes that inherit from std::enable_shared_from_this please make the constructors private and provide a factory function that returns shared_ptr